Automattic/jetpack
# Jetpack Sync: Add the previous interval end to full sync posts, comments

#### Changes proposed in this Pull Request:


For clarification purposes note that "posts and commented" include all the different `post_types` (post, pages, revisions, attachments, we actually have a blacklist for this) and `comment types` (comments, ping, trackbacks) that we care about. 

* Adds the previous interval end will allow us to know what the range is in between sync batches allowing us to safely delete the posts during a full sync that do not fall under the full range.

Previously we were sending batch posts and comments during full sync like this. 
`$post_ids = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];`

When we did a full sync we would send the ids in the following order. 

`[12, 11, 10, 9, 8, 7, 6, 5, 4, 3] , [ 2, 1 ] `

Which then would be expanded to the whole post object. 

`[ [ {ID:12, post_title:...}, {ID:12, post_title:...}, ...], [post_meta], [taxonomies] ]`

This PR changes things so that now we also send the previous interval end

`[ [ {ID:12, post_title:...}, {ID:11 post_title:...}, ..., {ID:3, post_title:...}], [post_meta], [taxonomies], *~0* ]`
and then on the next full sync of posts. we would send 
`[ [ {ID:2, post_title:...}, {ID:1, post_title:...} ], [post_meta], [taxonomies], *3* ]`

The same applies to comments. 
Knowing the previous interval end will allows us to determine the full range that we can use to potentially remove deleted posts, comments, and users. 


#### Testing instructions:
- Do the test pass?
- Do a full sync, do full sync post events send the previous interval end as expected?
 - Point your jetpack site to your dotcom sandbox. `define( 'JETPACK__API_BASE', 'https://somethingsomething.wordpress.com/jetpack.' );`
 -  Add the following in your `0-sandbox.php` 
```
// Remote action
add_action( 'jetpack_sync_remote_action', 'testing_sync_log_sync', 10, 9 );

function testing_sync_log_sync( $action_name, $args, $user_id, $silent, $timestamp, $sent_timestamp, $queue_id, $token, $actor ) {
	l( $action_name .  ' - blog_id: ' .$token->blog_id )
        l(  $args );
}
```


#### Proposed changelog entry for your changes:

*
